sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/session/HiveSessionImpl.java

```diff
@@ -141,7 +141,8 @@ public void open(Map<String, String> sessionConfMap) throws HiveSQLException {
     sessionState = new SessionState(hiveConf, username);
     sessionState.setUserIpAddress(ipAddress);
     sessionState.setIsHiveServerQuery(true);
-    SessionState.start(sessionState);
+    // Use setCurrentSessionState to avoid creating useless SessionDirs.
+    SessionState.setCurrentSessionState(sessionState);
```

Comment:
Hm, actually @wangyum. I just noticed that this is different from Hive's https://github.com/apache/hive/blob/master/service/src/java/org/apache/hive/service/cli/session/HiveSessionImpl.java#L180.

Shouldn't we at least keep:

https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java#L692-L699

for session history?

Reply:
It would be great if we can justify line by line. I think we won't have to take care about Tez but what about others here:

https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java#L688-L766

?

Reply:
We have disabled the session history:
https://github.com/apache/spark/blob/ebb4858f7185c6525adc4b23bc89f0a8262bf940/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L1271-L1276

Reply:
Thanks Yumming. What about other codes? I know we don't have to take care about Tez but wonder if it's safe for other codes there.

Reply:
Other codes also do not needed. We have disabled creating these directories for more than a year with the following changes:
![image](https://user-images.githubusercontent.com/5399861/116785447-312cc500-aacc-11eb-8dff-6ae75fbbc4d7.png)

Reply:
@wangyum, last question. What about this? Is it safe to remove?

```
      // Hive object instance should be created with a copy of the conf object. If the conf is
      // shared with SessionState, other parts of the code might update the config, but
      // Hive.get(HiveConf) would not recognize the case when it needs refreshing
      Hive.get(new HiveConf(startSs.conf)).getMSC();
```

Reply:
It has removed by https://github.com/apache/hive/commit/9bebaf6196c1842c285bfbad50765170e064f6e4.

![image](https://user-images.githubusercontent.com/5399861/121709709-67b62080-cb0b-11eb-9887-ed43258acd1c.png)